docs: Divide docs in globla referene and reference per exchange

pcriadoperez opened the pull request:
- Seperates exchange specific docs into it's own page to increase performance of the docs
- Create two sets of templates, one for the global api spec and one for the exchange specific docs
- Jsdocsmd generates the sidebar menu
- Added generated files to gitignore.
- I commit the generated files in this PR once so it gets created, but are part of gitignore for the future


// TODO: in the future we should change the links of all the docs to their specific exchange instead of the global one.

![image](https://github.com/ccxt/ccxt/assets/12142844/3eccabb7-a11f-47cf-8aa5-3c72b30214a5)

carlosmiei commented:
@pcriadoperez question, isn't this going to make the page very heavy?

carlosmiei commented:
> > @pcriadoperez question, isn't this going to make the page very heavy?
> 
> The way around, wouldn't this make it lighter? We are making the spec.md file much smaller and dividing into different pages. At least locally it seemed to be faster.

@pcriadoperez fair enough 😅,

Also, do we need to commit the generated files? Can't they be generated by the pipeline?

pcriadoperez commented:
> > > @pcriadoperez question, isn't this going to make the page very heavy?
> > 
> > 
> > The way around, wouldn't this make it lighter? We are making the spec.md file much smaller and dividing into different pages. At least locally it seemed to be faster.
> 
> @pcriadoperez fair enough 😅,
> 
> Also, do we need to commit the generated files? Can't they be generated by the pipeline?

Not really, I removed them and added a .gitkeep so the folder does get commited
